lib/_emerge/depgraph.py

```diff
@@ -9157,6 +9162,47 @@ def display_problems(self):
 			show_mask_docs()
 			writemsg("\n", noiselevel=-1)
 
+		# Build a list of problematic packages
+		maintainer_nag_packages = []
+
+		for pkg in self._dynamic_config.digraph:
+			if not isinstance(pkg, Package):
+				continue
+
+			maintainer_nag = pkg.root_config.settings.repositories[pkg.repo].maintainer_nag
+			maintainer_nag_url = pkg.root_config.settings.repositories[pkg.repo].maintainer_nag_url
+			if not maintainer_nag:
+				continue
+
+			path = os.path.dirname(pkg.root_config.trees['porttree'].dbapi.findname(pkg.cpv, myrepo=pkg.repo))
```

Comment:
We need to check if `pkg.type_name == 'ebuild'` before doing this, in order to avoid the test failure seen here: 

https://travis-ci.org/github/gentoo/portage/builds/665647387

Reply:
I noticed this and was checking the findname output for None, but this is nicer. I'll change to that.